### ISSUE TITLE
`unnecessary_parenthesis` with binary comparison

### ISSUE THREAD

**Issue by asashour** [dart-lang/linter#4063]
```dart
bool f(int x) => x == (10);
```

Expected:
[unnecessary_parenthesis](https://dart-lang.github.io/linter/lints/unnecessary_parenthesis.html) to trigger.

Actual:
No lint

**Comment by bwilkerson** [comment]
This has the potential for breaking existing users.
